tools/ping/src/lib.rs

```diff
@@ -14,8 +17,18 @@ use std::pin::Pin;
 
 pub mod cli;
 mod csv;
+mod metrics;
 
-pub use cli::PingCommand;
+async fn prometheus_handler() -> Result<HttpResponse, HttpError> {
```

Comment:
btw could also just use `near_jsonrpc::prometheus_handler` instead of copying it here, since it's marked `pub`:

```diff
diff --git a/Cargo.lock b/Cargo.lock
index e2d2ce825..75a4ec78a 100644
--- a/Cargo.lock
+++ b/Cargo.lock
@@ -3361,11 +3361,11 @@ dependencies = [
  "anyhow",
  "chrono",
  "clap 3.1.18",
+ "near-jsonrpc",
  "near-network",
  "near-o11y",
  "near-primitives",
  "once_cell",
- "prometheus",
  "tokio",
  "tracing",
 ]
diff --git a/tools/ping/Cargo.toml b/tools/ping/Cargo.toml
index 470eee863..c12d10991 100644
--- a/tools/ping/Cargo.toml
+++ b/tools/ping/Cargo.toml
@@ -12,10 +12,10 @@ anyhow.workspace = true
 chrono.workspace = true
 clap.workspace = true
 once_cell.workspace = true
-prometheus.workspace = true
 tokio.workspace = true
 tracing.workspace = true
 
+near-jsonrpc = { path = "../../chain/jsonrpc" }
 near-network = { path = "../../chain/network" }
 near-o11y = { path = "../../core/o11y" }
 near-primitives = { path = "../../core/primitives" }
diff --git a/tools/ping/src/lib.rs b/tools/ping/src/lib.rs
index 14b572118..faf55deb4 100644
--- a/tools/ping/src/lib.rs
+++ b/tools/ping/src/lib.rs
@@ -1,4 +1,4 @@
-use actix_web::{web, App, Error as HttpError, HttpResponse, HttpServer};
+use actix_web::{web, App, HttpServer};
 use anyhow::Context;
 pub use cli::PingCommand;
 use near_network::raw::{ConnectError, Connection, ReceivedMessage};
@@ -8,7 +8,6 @@ use near_primitives::hash::CryptoHash;
 use near_primitives::network::PeerId;
 use near_primitives::types::{AccountId, BlockHeight, EpochId};
 use near_primitives::version::ProtocolVersion;
-use prometheus::{Encoder, TextEncoder};
 use std::cmp;
 use std::collections::hash_map::Entry;
 use std::collections::{BTreeMap, BTreeSet, HashMap, HashSet};
@@ -19,17 +18,6 @@ pub mod cli;
 mod csv;
 mod metrics;
 
-async fn prometheus_handler() -> Result<HttpResponse, HttpError> {
-    let mut buffer = vec![];
-    let encoder = TextEncoder::new();
-    encoder.encode(&prometheus::gather(), &mut buffer).unwrap();
-
-    match String::from_utf8(buffer) {
-        Ok(text) => Ok(HttpResponse::Ok().body(text)),
-        Err(_) => Ok(HttpResponse::ServiceUnavailable().finish()),
-    }
-}
-
 // TODO: also log number of bytes/other messages (like Blocks) received?
 #[derive(Debug, Default)]
 struct PingStats {
@@ -448,7 +436,9 @@ async fn ping_via_node(
     tokio::pin!(next_timeout);
 
     let server = HttpServer::new(move || {
-        App::new().service(web::resource("/metrics").route(web::get().to(prometheus_handler)))
+        App::new().service(
+            web::resource("/metrics").route(web::get().to(near_jsonrpc::prometheus_handler)),
+        )
     })
     .bind(prometheus_addr)
     .unwrap()
```